app/models/manageiq/providers/ibm_cloud/power_virtual_servers/cloud_manager/image_import_workflow.rb

```diff
@@ -0,0 +1,111 @@
+class ManageIQ::Providers::IbmCloud::PowerVirtualServers::CloudManager::ImageImportWorkflow < ManageIQ::Providers::AnsiblePlaybookWorkflow
+  def load_transitions
+    self.state ||= 'initialize'
+
+    {
+      :initializing      => {'initialize'         => 'waiting_to_start'},
+      :start             => {'waiting_to_start'   => 'pre_execute'},
+      :pre_execute       => {'pre_execute'        => 'execute'},
+      :execute           => {'execute'            => 'running'},
+      :poll_runner       => {'running'            => 'running'},
+      :post_execute      => {'running'            => 'post_execute_poll'},
+      :post_execute_poll => {'post_execute_poll'  => 'post_execute_poll'},
+      :finish            => {'*'                  => 'finished'},
+      :abort_job         => {'*'                  => 'aborting'},
+      :cancel            => {'*'                  => 'canceling'},
+      :error             => {'*'                  => '*'}
+    }
+  end
+
+  def post_execute
+    ems = ExtManagementSystem.find(options[:ems_id])
+
+    body = {
+      :source        => 'url',
+      :imageName     => options[:miq_img][:name],
+      :osType        => options[:miq_img][:os],
+      :diskType      => options[:diskType],
+      :imageFilename => "#{options[:session_id]}.ova",
+      **options[:cos_pvs_creds]
+    }
+
+    ems.with_provider_connection(:service => 'PCloudImagesApi') do |api|
+      response = api.pcloud_cloudinstances_images_post(ems.uid_ems, body, {})
```

Comment:
What is the playbook doing?  Could this post and the subsequent poll be handled there?  That way the core poll_runner state would just work

Reply:
Technically it can be and I had the implementation for it somewhere at the early stages but our team voted against such design. One reason being that I'd be duplicating REST calls in my Ansible / Python code, which are already available in our ruby gem and thus should be reused. Another reason is that we'd have to unnecessarily provide PowerVS credentials to the Ansible scripts running on a powervc node (which can be thus avoided).